pkg/kubelet/stability/manager_linux.go

```diff
@@ -0,0 +1,287 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package stability
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/golang/glog"
+	"github.com/google/cadvisor/info/v2"
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/resource"
+	"k8s.io/kubernetes/pkg/apis/componentconfig"
+	"k8s.io/kubernetes/pkg/kubelet/cadvisor"
+	qosutil "k8s.io/kubernetes/pkg/kubelet/qos/util"
+	"k8s.io/kubernetes/pkg/util/wait"
+)
+
+// evictionOptionsImpl implements the EvictionOptions interface
+type evictionOptionsImpl struct {
+	podLifecycleManager PodLifecycleManager
+	resourceAccountant  ResourceAccountant
+	observations        []EvictionObservation
+}
+
+// GetPods returns the list of candidate pods to evict.
+// TODO: filter this list
+func (e *evictionOptionsImpl) GetPods() []*api.Pod {
+	return e.podLifecycleManager.GetPods()
+}
+
+// Accountant knows how to measure pod resource usage to help make an eviction decision.
+func (e *evictionOptionsImpl) Accountant() ResourceAccountant {
+	return e.resourceAccountant
+}
+
+// Observations are the list of eviction observations made that trigged an eviction action.
+func (e *evictionOptionsImpl) Observations() []EvictionObservation {
+	return e.observations
+}
+
+// ensure it implements required interface
+var _ EvictionOptions = &evictionOptionsImpl{}
+
+// resourceAccountantImpl implements ResourceAccountant using cadvisor
+type resourceAccountantImpl struct {
+	// used to collect usage information around a pod
+	cadvisorInterface cadvisor.Interface
+}
+
+// PodUsageInfo returns usage information for each container in a pod.
+func (r *resourceAccountantImpl) PodUsageInfo(pod *api.Pod) (api.ResourceList, error) {
```

Comment:
Need to move this to use `stats.PodStats`

https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/server/stats/summary.go#L185